### PR TITLE
add get_curve

### PR DESCRIPTION
**Pull request by ehaydenr**

### CODE REVIEW

**Comment by rushilmehra** [comment]
Just needs a rebase, we can remove the `#[cfg(not(feature = "kx-safe-default"))]` gate on `get_curve`

**Review by nox — APPROVED** [review]

File: boring/src/ssl/test/mod.rs [hunk 1]
```diff
@@ -920,6 +920,16 @@ fn server_set_default_curves_list() {
     ssl.server_set_default_curves_list();
 }
 
+#[cfg(not(feature = "kx-safe-default"))]
```
Comment (nox) [inline boring/src/ssl/test/mod.rs:923]: No need for this attribute anymore.

```suggestion
```

File: boring/src/ssl/mod.rs [hunk 2]
```diff
@@ -2748,6 +2764,21 @@ impl SslRef {
             .expect("invalid default server curves list");
     }
 
+    /// Returns the [`SslCurve`] used for this `SslRef`.
+    ///
+    /// This corresponds to [`SSL_get_curve_id`]
+    ///
+    /// [`SSL_get_curve_id`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_get_curve_id
+    #[cfg(not(feature = "kx-safe-default"))]
+    pub fn get_curve(&self) -> Option<SslCurve> {
```
Comment (nox) [inline boring/src/ssl/mod.rs:2773]: Other getters don't use a `get` prefix.

```suggestion
    pub fn curve(&self) -> Option<SslCurve> {
```

File: boring/src/ssl/mod.rs [hunk 3]
```diff
@@ -2748,6 +2764,21 @@ impl SslRef {
             .expect("invalid default server curves list");
     }
 
+    /// Returns the [`SslCurve`] used for this `SslRef`.
+    ///
+    /// This corresponds to [`SSL_get_curve_id`]
+    ///
+    /// [`SSL_get_curve_id`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_get_curve_id
+    #[cfg(not(feature = "kx-safe-default"))]
```
Comment (nox) [inline boring/src/ssl/mod.rs:2772]: No need for this attribute anymore.

```suggestion
```

**Review by nox — CHANGES_REQUESTED** [review]

File: boring/src/ssl/test/mod.rs [hunk 4]
```diff
@@ -920,6 +920,15 @@ fn server_set_default_curves_list() {
     ssl.server_set_default_curves_list();
 }
 
+#[test]
+fn get_curve() {
+    let server = Server::builder().build();
+    let client = server.client_with_root_ca();
+    let client_stream = client.connect();
+    let curve = client_stream.ssl().get_curve().expect("curve");
```
Comment (nox) [inline boring/src/ssl/test/mod.rs:928]:
```suggestion
    let curve = client_stream.ssl().curve().expect("curve");
```

**Review by nox — APPROVED** [review]

File: boring/src/ssl/mod.rs [hunk 5]
```diff
@@ -699,6 +699,22 @@ impl SslCurve {
...
+
+    /// Returns the curve name
+    ///
+    /// This corresponds to [`SSL_get_curve_name`]
+    ///
+    /// [`SSL_get_curve_name`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_get_curve_name
+    pub fn name(&self) -> Option<&'static str> {
+        unsafe {
+            let ptr = ffi::SSL_get_curve_name(self.0 as u16);
+            if ptr == std::ptr::null() {
```
Comment (nox) [inline boring/src/ssl/mod.rs:711]:
```suggestion
            if ptr.is_null() {
```

File: boring/src/ssl/mod.rs [hunk 6]
```diff
@@ -2748,6 +2764,20 @@ impl SslRef {
...
+    ///
+    /// This corresponds to [`SSL_get_curve_id`]
+    ///
+    /// [`SSL_get_curve_id`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_get_curve_id
+    pub fn curve(&self) -> Option<SslCurve> {
+        let curve_id = unsafe { ffi::SSL_get_curve_id(self.as_ptr()) };
+        if curve_id == 0 {
+            return None;
+        }
+        let curve_id: c_int = curve_id.try_into().ok()?;
```
Comment (nox) [inline boring/src/ssl/mod.rs:2777]: Clippy complains about this, is `.into()` doable on all archs or should we silence the lint there?

Comment (ehaydenr) [reply boring/src/ssl/mod.rs]: I imagine it is. Updated

**Review by rushilmehra — APPROVED** [review]

**Pull request merged**